engine/api/workflow/execute_node_job_run.go

```diff
@@ -123,20 +123,29 @@ func prepareSpawnInfos(j *sdk.WorkflowNodeJobRun, infos []sdk.SpawnInfo) error {
 
 // TakeNodeJobRun Take an a job run for update
 func TakeNodeJobRun(db gorp.SqlExecutor, store cache.Store, p *sdk.Project, id int64, workerModel string, workerName string, workerID string, infos []sdk.SpawnInfo, chanEvent chan<- interface{}) (*sdk.WorkflowNodeJobRun, error) {
-	job, err := LoadAndLockNodeJobRunWait(db, store, id)
+	// first load without FOR UPDATE WAIT to quick check status
```

Comment:
if we want to check the status, maybe we should only make a precise query to get only the node status without all the node structure